pkg/querier/stats/stats.go

```diff
@@ -193,6 +285,12 @@ func (s *QueryStats) Merge(other *QueryStats) {
 	s.AddFetchedSamples(other.LoadFetchedSamples())
 	s.AddFetchedChunks(other.LoadFetchedChunks())
 	s.AddExtraFields(other.LoadExtraFields()...)
+	s.AddQuery(other.LoadQuery())
+	s.AddQueryType(other.LoadQueryType())
+	s.AddStart(other.LoadStart())
+	s.AddEnd(other.LoadEnd())
+	s.AddStep(other.LoadStep())
+	s.AddTs(other.LoadTs())
```

Comment:
I think we can remove these. `Merge` is used for those stats processed through querier. But for query details we know it is only set in query frontend so we will do it once only.

Reply:
Got it. Removed those.